app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,71 @@
+class FoodsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_food, only: %i[show edit update destroy]
+
+  # GET /foods or /foods.json
+  def index
+    @foods = Food.all
+  end
+
+  # GET /foods/1 or /foods/1.json
+  def show; end
+
+  # GET /foods/new
+  def new
+    @food = Food.new
+  end
+
+  # GET /foods/1/edit
+  def edit; end
+
+  # POST /foods or /foods.json
+  def create
+    @food = Food.new(food_params)
+    @food.user_id = current_user.id
+
+    respond_to do |format|
+      if @food.save
+        format.html { redirect_to user_food_path(current_user.id, @food.id), notice: 'Food was successfully created.' }
+        format.json { render :show, status: :created, location: @food }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @food.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /foods/1 or /foods/1.json
+  # def update
+  #   respond_to do |format|
+  #     if @food.update(food_params)
+  #       format.html { redirect_to food_url(@food), notice: "Food was successfully updated." }
+  #       format.json { render :show, status: :ok, location: @food }
+  #     else
+  #       format.html { render :edit, status: :unprocessable_entity }
+  #       format.json { render json: @food.errors, status: :unprocessable_entity }
+  #     end
+  #   end
+  # end

```

Comment:
- [ ] it would be better to remove the unused commented to keep your files clean and pro